src/Chapter3.hs

```diff
@@ -907,9 +1001,23 @@ Implement instances of "Append" for the following types:
   ✧ *(Challenge): "Maybe" where append is appending of values inside "Just" constructors
 
 -}
+
+newtype Gold = Gold Int
 class Append a where
     append :: a -> a -> a
 
+instance Append Gold where 
+    append :: Gold -> Gold -> Gold
+    append (Gold a) (Gold b) = Gold (a+b)
+
+instance Append [a] where 
+    append :: [a] -> [a] -> [a]
+    append = (++)
+
+instance (Append a) => Append (Maybe a) where
+    append :: Maybe a -> Maybe a -> Maybe a
+    append (Just x) (Just y) = Just (append x y)
+    append _ _ = Nothing
```

Comment:
While this is valid implementation and nothing in the task description says differently, in reality it would be more beneficial is this function returned `Just` when at least one argument is `Just`.